graph2/graph.proto

```diff
@@ -0,0 +1,307 @@
+syntax = "proto3";
+package graph2;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "sourcegraph.com/sqs/pbtypes/timestamp.proto";
+import "sourcegraph.com/sourcegraph/srclib/toolref.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+
+// GitHubRepo holds additional metadata about GitHub repos.
+message GitHubRepo {
+	int32 stars = 1;
+}
+
+// RepoPermissions describes the possible permissions that a user (or an anonymous
+// user) can be granted to a repository.
+message RepoPermissions {
+	bool read = 1;
+	bool write = 2;
+	bool admin = 3;
+}
+
+message TreeKey {
+	// TreeType is the type of source tree.
+	// (git, hg, build-system specific like pip or npm, ftp)
+	string tree_type = 1;
+
+	// URI uniquely identifies a source tree among the universe of
+	// source trees of the same type. It also typically encodes how the
+	// source tree should be fetched.
+	string uri = 2 [(gogoproto.customname) = "URI"];
```

Comment:
It will typically be a URI/URL-like thing, like "github.com/gorilla/mux" or "hg.python.org/cpython" or "pypi.python.org/urllib3". Are there non-URI IDs you're thinking of?